### PR TITLE
Enhancement Updated PasswordValidator to fallback to config options (ss5)

### PR DESCRIPTION
**Pull request by flamerohr**
This pull request goes to the `master` (pseudo-`5`) branch and removes the instance variables retained in https://github.com/silverstripe/silverstripe-framework/pull/7800

For more details see: https://github.com/silverstripe/silverstripe-framework/issues/7724

### CODE REVIEW

**Comment by dhensby** [comment]
test failures need addresing

File: src/Security/PasswordValidator.php [hunk 1]
```diff
@@ -3,52 +3,65 @@
...
+     * @deprecated 5.0
      * Minimum password length
      *
      * @param int $minLength
      * @return $this
      */
     public function minLength($minLength)
     {
-        $this->minLength = $minLength;
+        Deprecation::notice('5.0', 'Use ->config()->set(\'min_length\', $value) instead.');
```
Comment (robbieaverill) [inline src/Security/PasswordValidator.php]: You've deprecated these in 4.x via https://github.com/silverstripe/silverstripe-framework/pull/7800 - should these be removals in the master branch instead?

**Comment by flamerohr** [comment]
Closing in favour of keeping the instance variables

File: src/Security/PasswordValidator.php (hunk 1 above)
Comment (flamerohr) [reply src/Security/PasswordValidator.php]: I'd remove them in 5.1 as a preference - 5.0 means that the method is highlighted with a notice
But yeah could swing either way.

**Pull request closed without merging**